SherpaBackEnd.Tests/Controllers/TeamMemberControllerTest.cs

```diff
@@ -0,0 +1,112 @@
+    {
+        var notSuccessfulAdding = new ConnectionToRepositoryUnsuccessfulException("Cannot perform add member function.");
+        _mockTeamMemberService.Setup(_ => _.AddTeamMemberToTeamAsync(It.IsAny<AddTeamMemberDto>())).ThrowsAsync(notSuccessfulAdding);
+
+        var allTeamMembers = await _teamMemberController.AddTeamMemberToTeamAsync(It.IsAny<AddTeamMemberDto>());
+        var resultObject = Assert.IsType<ObjectResult>(allTeamMembers);
+        Assert.Equal(StatusCodes.Status500InternalServerError, resultObject.StatusCode);
+    }
+
+    [Fact]
+    public async Task ShouldCallGetAllTeamMemberByTeamAsyncFromService()
+    {
+        var teamId = Guid.NewGuid();
+
+        await _teamMemberController.GetAllTeamMembersAsync(teamId);
+
+        _mockTeamMemberService.Verify(_ => _.GetAllTeamMembersAsync(teamId), Times.Once);
+    }
+
+    [Fact]
+    public async Task ShouldReturnOkWhenEmptyTeamListRetrievedWhileGettingAllTeams()
+    {
+        var teamId = Guid.NewGuid();
+        var member1Id = Guid.NewGuid();
+        var teamMember1 = new TeamMember(member1Id, "Name1", "Position1", "email1@gov.com");
+        var member2Id = Guid.NewGuid();
+        var teamMember2 = new TeamMember(member2Id, "Name2", "Position2", "email2@gov.com");
+
+        var teamMemberList = new List<TeamMember>() { teamMember1, teamMember2 };
+
+        _mockTeamMemberService.Setup(service => service.GetAllTeamMembersAsync(teamId))
+            .ReturnsAsync(teamMemberList);
+
+        var getAllTeamsAction = await _teamMemberController.GetAllTeamMembersAsync(teamId);
+        var okObjectResult = Assert.IsType<OkObjectResult>(getAllTeamsAction.Result);
+        var teamMembers = Assert.IsType<List<TeamMember>>(okObjectResult.Value);
+        Assert.True(teamMembers.Any());
+    }
+
+    [Fact]
+    public async Task ShouldReturnNotFoundWhenGetTeamByIdAsyncCanNotFindThatTeamId()
+    {
+        var teamId = Guid.NewGuid();
+        _mockTeamMemberService.Setup(_ => _.GetAllTeamMembersAsync(teamId)).ReturnsAsync((IEnumerable<TeamMember>)null);
+
+        var allTeamMembers = await _teamMemberController.GetAllTeamMembersAsync(teamId);
+        var notFoundResult = Assert.IsType<NotFoundResult>(allTeamMembers.Result);
+        Assert.Equal(StatusCodes.Status404NotFound, notFoundResult.StatusCode);
+    }
+
+    [Fact]
+    public async Task ShouldReturnErrorIfATeamCanNotBeFoundByTheTeamId()
+    {
+        var teamId = Guid.NewGuid();
+        var notSuccessfulGetting = new ConnectionToRepositoryUnsuccessfulException("Cannot perform get team function.");
+        _mockTeamMemberService.Setup(_ => _.GetAllTeamMembersAsync(teamId)).ThrowsAsync(notSuccessfulGetting);
+
+        var allTeamMembers = await _teamMemberController.GetAllTeamMembersAsync(teamId);
+        var resultObject = Assert.IsType<ObjectResult>(allTeamMembers.Result);
+        Assert.Equal(StatusCodes.Status500InternalServerError, resultObject.StatusCode);
+    }
```

Comment:
This is a bit strange, I would go with a simple uncatched general Exception, cause all of them should cause a 500 status code